src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt

```diff
@@ -91,15 +92,68 @@ class VisitNotificationEventService(
   }
 
   private fun processVisitsWithNotifications(affectedVisits: List<VisitDto>, type: NotificationEventType) {
+    val affectedVisitsNoDuplicate = affectedVisits.filter { !visitNotificationEventRepository.isEventARecentDuplicate(it.reference, type) }
+    flagTrackEvents(affectedVisitsNoDuplicate, type)
```

Comment:
should we track this event only after we save to DB?

Reply:
good point,...

Reply:
After talk with @dhirajshettymoj decided to leave as is since this may be taken out and also report should consistent format no matter what the event type